packages/central-server/app/sync/CentralSyncManager.js

```diff
@@ -98,13 +98,22 @@ export class CentralSyncManager {
     const parameters = { deviceId, facilityIds, isMobile };
 
     const unmarkSessionAsProcessing = await this.markSessionAsProcessing(sessionId);
-    const syncSession = await this.store.models.SyncSession.create({
-      id: sessionId,
-      startTime,
-      lastConnectionTime: startTime,
-      debugInfo,
-      parameters,
-    });
+    let syncSession;
+    try {
+      syncSession = await this.store.models.SyncSession.create({
+        id: sessionId,
+        startTime,
+        lastConnectionTime: startTime,
+        debugInfo,
+        parameters,
+      });
+    } catch (error) {
+      // If the session creation fails, we need to release the lock otherwise it will hold up
+      // an open connection until the application restarts
+      await unmarkSessionAsProcessing();
+      error.message = `Failed to create sync session, server may be overloaded with sync requests: ${error.message}`;
+      throw error;
+    }
```

Comment:
These are the only changes, the rest of the delta are just prettier doing its thing.

So there's a few different ways to approach this. Possibly the easiest would have been to move the `markSessionAsProcessing()` call until after we create the session. However I kind of recall us wanting to avoid that, might need to think it through more.

Decided to go with this try/catch and add a more helpful error message.

There's also an argument that we should really consider bumping the default connection pool size. It's set to 10, and given that every sync can hold open 2 connections, and pub-sub holds at least one more, we're pretty close to maxing out (we support 4 concurrent syncs by default).